pkg/karmadactl/promote.go

```diff
@@ -159,6 +159,18 @@ func (o *CommandPromoteOption) Complete(args []string) error {
 		}
 	}
 
+	if o.Namespace == "" {
```

Comment:
this check is redundant.

FYI https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/apply/apply.go#L292

Reply:
We don't use the `flags.Factory.ToRawKubeConfigLoader().Namespace()` directly, it is the default namespace in kubeconfig file, if we need to set other namespace, we need to speficy it.

https://github.com/karmada-io/karmada/blob/9b70cf2873a1dbd5d3ed45094e85e2b317b1ba44/pkg/karmadactl/promote.go#L324

Reply:
Because, the namespace is not necessary for all subcommands, so we could extract the global flags `kubeconfig` and `karmada-context`. refer https://github.com/karmada-io/karmada/pull/2684#discussion_r1007801205

How do you think? @carlory

Reply:
There is a significant difference between karmadactl and kubectl, kubectl operates on resources and is not responsible for cluster creation, while karmadactl is not. For deployment-related commands, it is indeed not suitable to introduce this parameter.

howerver, on the other hand, openshift has a command binary named oc, which can not only create a simple openshift cluster, but also operate the cluster. oc takes namespace as a global parameter.

IMO, I can accept them both. If the installation logic is splited from the karmadactl in the feature, I intend to takes namespace as a global parameter.

@lonelyCZ

Reply:
> IMO, I can accept them both. If the installation logic is splited from the karmadactl in the feature, I intend to takes namespace as a global parameter.

Yes,  at that point, we can do it. It is clearer to separate temporarily and avoid invalid options.